integrations/kube-agent-updater/pkg/cache/cache.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/gravitational/trace"
+)
+
+// StringAndErrorCache wraps a function returning (string, error) and caches the
```

Comment:
I've seen that there's a dependent on this that's essentially storing a bool as a string and then reconverting it to a bool each time. What if we used generics here to avoid this awkward conversion?

```go
type Cache[T any] struct {
    cachedValue T
    getter func(context.Context) (T, error)
}
```

Reply:
Done